### DISCUSSION TITLE
Achieving a feature of mpv.net

### DISCUSSION THREAD

**Discussion by sqq135** [tomasklaen/uosc#840] [General]
I think the mpv.net's feature, that the OSC automatically hides, even when the mouse is within the functional area (e.g., over the timeline bar or thumbfast display) and not in motion, is very convenient. But I don't know how to achieve this feature in your uosc. (mpv.net has a terrible subtitle system, so I replace it by another mpv fork, hooke007/MPV_lazy, using your uosc)

I study the portable_config/script-opts/osc.conf, but I couldn't find the correct parameter. I'm really not good at coding, hoping you can give me some guidance.

**Comment by christoph-heinrich** [comment]
```diff
diff --git a/src/uosc/lib/cursor.lua b/src/uosc/lib/cursor.lua
index c1600ed..223e9c0 100644
--- a/src/uosc/lib/cursor.lua
+++ b/src/uosc/lib/cursor.lua
@@ -259,7 +259,6 @@ function cursor:leave() self:move(math.huge, math.huge) end
 
 function cursor:is_autohide_allowed()
 	return options.autohide and (not self.autohide_fs_only or state.fullscreen) and
-		not (self.mbtn_left_dbl_enabled or self.mbtn_right_enabled or self.wheel_enabled) and
 		not Menu:is_open()
 end
 mp.observe_property('cursor-autohide-fs-only', 'bool', function(_, val) cursor.autohide_fs_only = val end)
```

**Reply by sqq135** [reply]
It doesn't work after copying your code and clean the cache. I think this isn't the right location after asking the GPT. (I really don't understand code) 

I think the original logic is that when the mouse move in an area (upper or lower parts), the osc would not hide unless the mouse out. 

But I need that when the mouse move in an area, the osc only show in few seconds and hide, keeping showing unless the mouse moves, or hiding at once when mouse moves out.

I'm only a outsider and don't know the level of difficulty to achieve the function. Hoping my nonsense has some inspiration for you.

**Reply by christoph-heinrich** [reply]
I'm failing to see how this didn't work for you.
1. find `cursor.lua` in your `~~/scripts/uosc` directory
2. delete the line that's marked red in the diff above

**Reply by Sneakpeakcss** [reply]
Probably because this also requires `autohide=yes` in `uosc.conf`. _Though, i'm somewhat interested in what's the "terrible subtitle system" in `mpv.net`, since from what i know it doesn't have any changes to it, and uses the same `mpv` default `sub-auto=exact` setting._
At first glance `MPV_lazy` seems to just [follow](https://github.com/hooke007/MPV_lazy/blob/e80b70eac0759febd5c8a950c27bb965cf395cb8/portable_config/mpv.conf#L250-L269) the default settings with additional font loading script, but, well… Getting a bit offtopic.

 I'm quite sure that the issue here is the `autohide` setting.

**Reply by christoph-heinrich** [reply]
Ah yes, I forgot about the `autohide=yes` :sweat_smile:

**Reply by sqq135** [reply]
Yes, it works. Thanks for your guidance.

I believe I should explain the reason for my low assessment about subtitle system; it was due to my misunderstanding and ignorance. In short, the fork I used caused this subtitle issue in the April 23 update.

After the prolonged cessation of update in mpv.net, its Chinese translation branch also stopped updating. The fork I used was based on it, and as a result, he had to update it himself, which led to the appearance of the defect. It's confusing that I couldn't find the relevant description in his update log.


https://github.com/hooke007/mpv.net_CM
https://github.com/hooke007/MPV_lazy/discussions/258

the fork I used, I like its shortcut and some concenient changes.
https://github.com/diana7127/mpv.net-DW/releases

**Reply by Sneakpeakcss** [reply]
Ahh, that makes sense. Looks like `mpv.net` had quite the turbulance, but came back to active development few months after that, and it even implemented localization support which was the mentioned issue. Though, at this point it's on a `v7` release, so that fork is quite outdated.

Chinese translation is natively supported now, and the initial work was provided by the author of another pre-configured [Chinese package](https://github.com/dyphire/mpv-config). I think both of them used `mpv.net` just for the configurable context menu, and not long ago dropped it(?) in favour of [mpv-menu-plugin](https://github.com/tsl0922/mpv-menu-plugin), which seems to be in full-development to the point that it might even become a default thing in `mpv` at some point.

The day where `mpv` comes with contextmenu out of box might be upon us (_well… At least on Windows_), isn't that unbelievable lol.

**Reply by hooke007** [reply]
>At first glance MPV_lazy seems to just [follow](https://github.com/hooke007/MPV_lazy/blob/e80b70eac0759febd5c8a950c27bb965cf395cb8/portable_config/mpv.conf#L250-L269) the default settings with additional font loading script, but, well… Getting a bit offtopic.

That's not my settings. The file in the main branch is a simplified manual.

>After the prolonged cessation of update in mpv.net, its Chinese translation branch also stopped updating.

Upstream has supported multi-languages.
